src/machine/machine_nrf.go

```diff
@@ -232,3 +232,98 @@ func (i2c I2C) readLastByte() byte {
 	i2c.signalStop() // signal 'stop' now, so it is sent when reading RXD
 	return byte(i2c.Bus.RXD)
 }
+
+// SPI on the NRF.
+type SPI struct {
+	Bus *nrf.SPI_Type
+}
+
+// There are 2 SPI interfaces on the NRF5x.
+var (
+	SPI0 = SPI{Bus: nrf.SPI0}
+	SPI1 = SPI{Bus: nrf.SPI1}
+)
+
+// SPIConfig is used to store config info for SPI.
+type SPIConfig struct {
+	Frequency uint32
+	SCK       uint8
+	MOSI      uint8
+	MISO      uint8
+	LSBFirst  bool
+	Mode      uint8
+}
+
+// Configure is intended to setup the SPI interface.
+func (spi SPI) Configure(config SPIConfig) {
+	// enable
+	spi.Bus.ENABLE = nrf.SPI_ENABLE_ENABLE_Disabled
+
+	// set frequency
+	var freq uint32
+
+	switch config.Frequency {
+	case 125000:
+		freq = nrf.SPI_FREQUENCY_FREQUENCY_K125
+	case 250000:
+		freq = nrf.SPI_FREQUENCY_FREQUENCY_K250
+	case 500000:
+		freq = nrf.SPI_FREQUENCY_FREQUENCY_K500
+	case 1000000:
+		freq = nrf.SPI_FREQUENCY_FREQUENCY_M1
+	case 2000000:
+		freq = nrf.SPI_FREQUENCY_FREQUENCY_M2
+	case 4000000:
+		freq = nrf.SPI_FREQUENCY_FREQUENCY_M4
+	case 8000000:
+		freq = nrf.SPI_FREQUENCY_FREQUENCY_M8
+	default:
+		freq = nrf.SPI_FREQUENCY_FREQUENCY_K500
+	}
+	spi.Bus.FREQUENCY = nrf.RegValue(freq)
+
+	var conf uint32
+
+	// set bit transfer order
+	if config.LSBFirst {
+		conf = (nrf.SPI_CONFIG_ORDER_LsbFirst << nrf.SPI_CONFIG_ORDER_Pos)
+	}
+
+	// set mode
+	switch config.Mode {
+	case 0:
```

Comment:
These modes appear to be magic constants. Can you give them descriptive names?
I don't know that much about SPI so I don't know what they should be called.
You can also use values like `nrf.SPI_CONFIG_CPOL_ActiveHigh << nrf.SPI_CONFIG_CPOL_Pos | nrf.SPI_CONFIG_CPHA_Leading << nrf.SPI_CONFIG_CPHA_Pos` when you define them, so that you can avoid the switch here. Or go even further and expose constants like `SPIModeActiveHigh` that should be ORed together by the user.

Reply:
Actually, they are known as "Mode 0", "Mode 1" etc. So I thought `Mode0 = 0` seemed redundant.

Reply:
Hmmm yeah apparently they don't have standard names. Then it's okay.